sycl/doc/extensions/experimental/sycl_ext_oneapi_cuda_tex_cache_read.asciidoc

```diff
@@ -98,7 +98,7 @@ T ldg(const T* ptr);
 `ldg` returns the data of type `T` located at address `ptr`. When called from the `ext_oneapi_cuda` backend the data is cached in the read-only texture cache.
 When called from any other backend a copy of the data stored at address `ptr` is returned without using any special cache.
 
-The template parameter `T` can be one of `char`, `signed char`, `short`, `int`, `long`, `long long`, `unsigned char`, `unsigned short`, `unsigned int`, `unsigned long`, `unsigned long long`, `vec<char, 2>`, `vec<char, 4>`, `vec<short, 2>`, `vec<short, 4>`, `vec<int, 2>`, `vec<int, 4>`, `vec<long long, 2>`, `vec<uchar, 2>`, `vec<uchar, 4>`, `vec<ushort, 2>`, `vec<ushort, 4>`, `vec<uint, 2>`, `vec<uint, 4>`, `vec<unsigned long long, 2>`, `float`, `vec<float, 2>`, `vec<float, 4>`, `double`, or `vec<double, 2>`.
+The template parameter `T` can be one of `char`, `signed char`, `short`, `int`, `long`, `long long`, `unsigned char`, `unsigned short`, `unsigned int`, `unsigned long`, `unsigned long long`, `half`, `float`, `double`, `vec<char, 2>`, `vec<char, 4>`, `vec<short, 2>`, `vec<short, 4>`, `vec<int, 2>`, `vec<int, 4>`, `vec<long long, 2>`, `vec<uchar, 2>`, `vec<uchar, 4>`, `vec<ushort, 2>`, `vec<ushort, 4>`, `vec<uint, 2>`, `vec<uint, 4>`, `vec<unsigned long long, 2>`, `vec<half, 2>`, `vec<float, 2>`, `vec<float, 4>`, or `vec<double, 2>`.
```

Comment:
Actually maybe half4/double4 are natively implementable, at least this is what is suggested by https://docs.nvidia.com/cuda/parallel-thread-execution/#data-movement-and-conversion-instructions-ld-global-nc
We could investigate this in the future.
The types that we added so far match those exposed through CUDA: https://docs.nvidia.com/cuda/cuda-c-programming-guide/#read-only-data-cache-load-function

Reply:
If this is what the CUDA docs say are natively implemented, that is fine.  I only pointed it out because it doesn't seem to match your existing pattern.  For the other types, it seems like there are 2- and 4-element versions for all the types that are 32-bits or less.  (E.g. float2, float4, int2, int4, short2, short4, etc.)

Reply:
Yes this matches the pattern here: https://docs.nvidia.com/cuda/cuda-c-programming-guide/#read-only-data-cache-load-function

However I don't see why we can't also implement it for additional types like double4 in the future, especially if the ptx allows this to be done via a single instruction. It is just that the equivalent of double4 etc is not exposed via CUDA Runtime API.

Reply:
If we are adding `vec<half, 4>` can you also add `vec<double, 4>`? MD uses the 4D version for double: https://github.com/glotzerlab/hoomd-blue/commit/730a8093a57d3b794c579d899a7cd63e90bac684#diff-c20b0af2267267f69ae103ae2c7e4aecea76b0e5feb3d3f22f0e48f4b652c08dR28

This will save them having to implement this themselves.

Thanks

Reply:
Sure, if we are sticking to vectors of 2 and 4 elements, I'll add 4 elements variants of `longlong` and `ulonglong`.